### ISSUE TITLE
Vitest/C8 coverage includes the unit test files themselves

### ISSUE THREAD

**Issue by lloydjatkinson** [vitest-dev/vitest#1612]
### Describe the bug

As per the docs I have the following npm script to generate the coverage report:

```
"coverage": "vitest run --coverage"
```

However it includes the unit test files themselves which is not useful and also leads to incorrect coverage metrics. Please can this be addressed in Vite?

![image](https://user-images.githubusercontent.com/5633938/177846489-a6464148-e792-4212-b565-a8ed3b63c622.png)
![image](https://user-images.githubusercontent.com/5633938/177846536-8916b1ce-f214-4b3e-8069-5d6fb305d92b.png)

### Reproduction

Follow getting started guide and make a basic function and a unit test that imports that function. Have `"coverage": "vitest run --coverage"` in npm scrips and run it. Observe the reports in both the console and HTML istanbul report contain both SUT's and the unit tests themselves.

### System Info

```shell
Windows and Linux, Node 16.
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by pandyarudra** [comment]
@lloydjatkinson Hey Lloyd, please try excluding the spec files in the coverage config options as follows: 
`// vite.config.js
export defineConfig({
  test: {
    coverage: {
      reporter: ['lcov', "html"],
      exclude: [
        '**/*.spec.ts',
      ],
    }
  }
});`


All you need is the **exclude** config.

**Comment by sheremet-va** [comment]
> All you need is the **exclude** config.

It should already be excluded. We have `defaults.ts` file with default coverage. We need to add `spec` alongside `test` there. PR welcome.

**Comment by seivan** [comment]
Which begs the question, what happens if you do in-source testing? I love that feature, and I can't leave it.

**Comment by OmgImAlexis** [comment]
> Which begs the question, what happens if you do in-source testing? I love that feature, and I can't leave it.

If this isn't fixed by #1627 I'd suggest making a new issue.
